hudi-hadoop-common/src/test/java/org/apache/hudi/common/table/TestHoodieTableConfig.java

```diff
@@ -266,15 +266,15 @@ public void testDropInvalidConfigs() {
     // test valid ones are not dropped
     config = new HoodieConfig();
     config.setValue(HoodieTableConfig.VERSION, String.valueOf(HoodieTableVersion.EIGHT.versionCode()));
-    config.setValue(RECORD_MERGE_MODE, RECORD_MERGE_MODE.defaultValue());
+    config.setValue(RECORD_MERGE_MODE, RECORD_MERGE_MODE.defaultValue().name());
     HoodieTableConfig.dropInvalidConfigs(config);
     assertTrue(config.contains(RECORD_MERGE_MODE));
   }
 
   @Test
   public void testDefinedTableConfigs() {
     List<ConfigProperty<?>> configProperties = HoodieTableConfig.definedTableConfigs();
-    assertEquals(37, configProperties.size());
+    assertEquals(36, configProperties.size());
```

Comment:
Should we add new tests around the logic of inferring the merge configs?

Reply:
addded